flang/include/flang/Parser/parse-tree.h

```diff
@@ -4008,6 +4015,13 @@ struct OmpMapClause {
   std::tuple<MODIFIERS(), OmpObjectList, bool> t;
 };
 
+// Ref: [5.2:217-218]
+// message-clause ->
+//    MESSAGE("message-text")
+struct OmpMessageClause {
+  WRAPPER_CLASS_BOILERPLATE(OmpMessageClause, std::string);
```

Comment:
The message doesn't have to be a string literal.  It needs to be known at compile time only if there is AT clause with COMPILATION as the argument.

Also, we should be able to accept this:
```
character(*), parameter :: s = "haha"
!$omp error at(compilation) message(s)
```

Reply:
Ok, so I have updated to allow pretty much anything there. I think we need some semantics checks to make sure it's not `cos(exp(pi * cmplx(3.0, -2.0))` or something, but for the purposes of compiler doesn't say "I have no idea what this means", it should work.